### PR TITLE
[BugFix] Fix query rewrite failed for external table after support view delta

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/2031

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 28 (32.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|9|00.00%|[397, 402, 403, 406, 407, 408, 409, 410, 411]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|10|00.00%|[122, 217, 222, 223, 226, 227, 228, 229, 230, 231]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|9|9|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java [hunk 1]
```diff
@@ -212,6 +212,26 @@ public boolean isSupported() {
         return true;
     }
 
+    @Override
+    public int hashCode() {
+        return com.google.common.base.Objects.hashCode(getCatalogName(), remoteDbName, getTableIdentifier());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java:217]: nit: import Objects

**Review by stephen-shelby — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**